src/rust/aggregator/sqlx_queries/candlesticks/update_last_indexed_txn_version.sql

```diff
@@ -0,0 +1,9 @@
+WITH parameters AS (
+    SELECT
+        $1::int AS resolution)
+UPDATE aggregator.candlesticks_last_indexed_txn
+SET
+    txn_version = COALESCE((SELECT (MAX(txn_version)) FROM fill_events), txn_version)
+FROM
```

Comment:
We're doing this in a transaction so this should not be a problem right ?

Reply:
It shouldn't be a problem if we use serializable transaction level for the aggregator run, because serializable prevents any phenomena between concurrent transactions: https://www.postgresql.org/docs/current/transaction-iso.html

But without explicitly specifying serializable transaction isolation level, the transaction (a) that inserts fill events and the transaction (b) that aggregates candlestick data could execute concurrently as follows:

1. Aggregator insert data function is called
2. Processor inserts fill events
3. Aggregator set last txn is called

This would valid in terms of concurrency collisions, but the result would be that some fill events are skipped over

Though I think there is a more elegant solution than setting isolation level serializable:

1. During aggregator candlestick run, query most recent fill event and store as variable in Rust loop
2. Pass that txn version to the insert data SQL code, and only index fill events up until that number
3. Then log that txn version as latest one indexed

This will enable maximum concurrency and also prevent concurrency anomalies

Reply:
I set the transaction isolation level to `REPEATABLE READ`, because I think your model still has concurrency issues. I'll also make a PR to add this to the leaderboards, as it should be there as well.

Reply:
Noting that the serialization anomaly on repeatable read has the effect that inserted fills will be correctly deferred until next cycle if concurrent transactions take place